src/fu-util.c

```diff
@@ -2393,6 +2639,60 @@ fu_util_security (FuUtilPrivate *priv, gchar **values, GError **error)
 		return FALSE;
 	str = fu_util_security_attrs_to_string (attrs);
 	g_print ("%s\n", str);
+
+	/* can we find a remote with a security attr */
+	remote = fu_util_get_remote_with_security_report_uri (priv, &error_local);
+	if (remote == NULL) {
+		g_debug ("failed to find suitable remote: %s", error_local->message);
+		return TRUE;
+	}
+
+	/* ask if we can upload them to the LVFS */
+	if (priv->no_unreported_check)
+		return TRUE;
+	if (!priv->assume_yes ||
+	    !fwupd_remote_get_automatic_security_reports (remote)) {
```

Comment:
But so don't you want to make sure it hasn't changed?  Otherwise users will upload every time they view security.

Reply:
I pondered this. I wondered if we should store this in the remote config file, or on the daemon, or somewhere else entirely. I'm somewhat regretting not returning metadata from the engine when getting the attr list. I guess we could just store the sha256 checksum in ~/cache somewhere, but it seems like the kind of thing the fwupd daemon should know.

Reply:
> I'm somewhat regretting not returning metadata from the engine when getting the attr list

This is why it's good 1.5.0 isn't released yet while figuring all this stuff out.  I don't see a problem with changing the functions and symbols right now.

>I guess we could just store the sha256 checksum in ~/cache somewhere, but it seems like the kind of thing the fwupd daemon should know.

I agree it should be kept in the daemon not home directory.  I was thinking all the attributes and the end result would both make sense to go into the history database.  A logical extension then ends up being for the client to look at the history of security attribute runs.

The client command `security` could potentially then indicate when security has gone up or down too.

Reply:
Thinking more about this, I don't think re-uploading the security report is such a bad thing. Some rationalle:

 * If the user got a low score (e.g. SB off, or non-encrypted swap) they might "fix" the issue and then re-get the security level. We definitely want the new report.

* If the user updates the distro/kernel and something like lockdown changes then we want the new report too.

* If the user updates the firmware, we definately want the new report as the IOMMU might have just turned up.

Having a "cache" even if a SHA256 of the blob means you have to know when to invalidate it.

Reply:
Re-uploading the security report in all of those scenarios makes perfect sense to me.  I don't disagree about re-uploading the report being bad, I just think it should only happen if it has changed in some way.

Reply:
Okay, I ruminated about this for over an hour, and the conclusion I came to was it's not worth adding the complexity. Sending the security report takes less than a second and is a few Kb of network, and is trivial to handle on the LVFS. It's literally the case of pressing enter to stop the action happening.

Reply:
I guess the part that I'm thinking about is if a graphical client that runs on every boot checks security and reads the value of the config file it also could be checking in every boot.  It would mean that the graphical client needs to cache the information and decide not to run it (which I guess might be acceptable - command line always sends and graphical checks each time),

Reply:
In that case I'd simply put the successful upload timestamp and sha256 in GSettings. We don't have per-user settings in fwupd at the moment.

Reply:
ok.